sdk/Sdk.Generators/FunctionExecutor/FunctionExecutorGenerator.Emitter.cs

```diff
@@ -0,0 +1,104 @@
+                         using Microsoft.Extensions.DependencyInjection;
+                         using Microsoft.Azure.Functions.Worker.Context.Features;
+                         using Microsoft.Azure.Functions.Worker.Invocation;
+                         namespace Microsoft.Azure.Functions.Worker
+                         {
+                             internal class DirectFunctionExecutor : IFunctionExecutor
+                             {
+                                 public async ValueTask ExecuteAsync(FunctionContext context)
+                                 {
+                                     {{GetMethodBody(functions)}}
+                                 }
+                             }
+                             public static class FunctionExecutorHostBuilderExtensions
+                             {
+                                 ///<summary>
+                                 /// Configures an optimized function executor to the invocation pipeline.
+                                 ///</summary>
+                                 public static IHostBuilder ConfigureGeneratedFunctionExecutor(this IHostBuilder builder)
+                                 {
+                                     return builder.ConfigureServices(s => 
+                                     {
+                                         s.AddSingleton<IFunctionExecutor, DirectFunctionExecutor>();
+                                     });
+                                 }
+                             }
+                         }
+                         """;
+
+            return result;
+        }
+
+        private static string GetMethodBody(IEnumerable<ExecutableFunction> functions)
+        {
+            var sb = new StringBuilder();
+            sb.Append(@"var inputBindingFeature = context.Features.Get<IFunctionInputBindingFeature>()!;
+            var inputBindingResult = await inputBindingFeature.BindFunctionInputAsync(context)!;
+            var inputArguments = inputBindingResult.Values;
+            ");
+            foreach (ExecutableFunction function in functions)
+            {
+                sb.Append($@"
+            if (string.Equals(context.FunctionDefinition.EntryPoint, ""{function.EntryPoint}"", StringComparison.OrdinalIgnoreCase))
+            {{");
+
+                int paramCounter = 0;
+                var constructorParamTypeNameList = new List<string>();
+                foreach (var argumentTypeName in function.ParentFunctionClass.ConstructorParameterTypeNames)
+                {
+                    paramCounter++;
+                    sb.Append($@"
+                var p{paramCounter} = context.InstanceServices.GetService<{argumentTypeName}>();");
+                    constructorParamTypeNameList.Add($"p{paramCounter}");
+                }
+                var constructorParamsStr = string.Join(", ", constructorParamTypeNameList);
+
+                int paramCounter2 = 0;
```

Comment:
nit-  could name `constructorParamCounter` and `functionParamCounter` ?

Reply:
Fixed. Thank you!